src/webgpu/api/validation/encoding/cmds/clearBuffer.spec.ts

```diff
@@ -40,6 +40,29 @@ g.test('invalid_buffer')
     });
   });
 
+g.test('buffer,device_mismatch')
+  .desc(`Tests clearBuffer cannot be called with buffer created from another device.`)
+  .paramsSubcasesOnly(u => u.combine('mismatched', [true, false]))
+  .fn(async t => {
+    const { mismatched } = t.params;
+
+    const bufferSize = 8;
+    const descriptor: GPUBufferDescriptor = {
+      size: bufferSize,
+      usage: GPUBufferUsage.COPY_DST,
+    };
+    const buffer = mismatched
+      ? t.getDeviceMismatchedBuffer(descriptor)
+      : t.createBufferWithState('valid', descriptor);
```

Comment:
nit: don't need to use createBufferWithState if the state is constant `'valid'`. Just directly use `t.device.createBuffer` which doesn't hide the API calls we're testing.

Same for the many other uses of createBufferWithState, createTextureWithState, and createQuerySetWithState in this PR.

Reply:
Here I want the buffer/texture/queryset could be tracked for cleanup by using `createBufferWithState`, `createTextureWithState`, and `createQuerySetWithState`.